react/src/components/AutoScalingRuleEditorModal.tsx

```diff
@@ -0,0 +1,320 @@
+      {...baiModalProps}
+      destroyOnClose
+      onOk={handleOk}
+      onCancel={handleCancel}
+      centered
+      title={
+        autoScalingRule
+          ? t('autoScalingRule.EditAutoScalingRule')
+          : t('autoScalingRule.AddAutoScalingRule')
+      }
+      confirmLoading={
+        isInflightAddAutoScalingRule || isInflightModifyAutoScalingRule
+      }
+    >
+      <Form
+        ref={formRef}
+        layout={'vertical'}
+        requiredMark={'optional'}
+        initialValues={
+          autoScalingRule
+            ? autoScalingRule
+            : {
+                metric_source: 'KERNEL',
+                comparators: 'LESS_THAN',
+                threshold: '10',
+                step_size: 0,
+                cooldown_seconds: 300,
+                min_replicas: 0,
+                max_replicas: 3,
+              }
+        }
+      >
+        <Form.Item
+          label={t('autoScalingRule.MetricName')}
+          name={'metric_name'}
+          rules={[{ required: true }]}
+        >
+          <Input />
+        </Form.Item>
+        <Form.Item
+          label={t('autoScalingRule.MetricSource')}
+          name={'metric_source'}
+          rules={[{ required: true }]}
+        >
+          <Select>
+            <Select.Option value={'INFERENCE_FRAMEWORK'}>
+              Inference Framework
+            </Select.Option>
+            <Select.Option value={'KERNEL'}>Kernel</Select.Option>
+          </Select>
+        </Form.Item>
+        <Flex
+          direction="row"
+          justify="around"
+          align="stretch"
+          style={{ width: '100%' }}
+          gap={'md'}
+        >
+          <Form.Item
+            label={t('autoScalingRule.Comparator')}
+            name={'comparator'}
+            rules={[{ required: true }]}
+            style={{ width: 600 }}
+          >
+            <Radio.Group
+              block
+              size={'small'}
+              options={comparators}
+              optionType={'button'}
+              buttonStyle={'solid'}
+            ></Radio.Group>
+          </Form.Item>
+          <Form.Item
+            label={t('autoScalingRule.Threshold')}
+            name={'threshold'}
+            rules={[{ required: true }]}
+            style={{ width: 200 }}
+          >
+            <Input></Input>
+          </Form.Item>
+        </Flex>
+        <Form.Item
+          label={'Step size'}
+          name={'step_size'}
+          rules={[{ required: true }]}
+        >
+          <InputNumberWithSlider
+            // TODO: hardcoded for now
+            min={-3}
+            max={3}
+            inputNumberProps={{}}
+            step={1}
+            allowNegative
+          />
+        </Form.Item>
+        <Form.Item
+          label={t('autoScalingRule.CoolDownSeconds')}
+          name={'cooldown_seconds'}
+          rules={[{ required: true }]}
+        >
+          <InputNumber defaultValue={300}></InputNumber>
+        </Form.Item>
+        <Flex
```

Comment:
`cooldown_seconds` cannot have a negative value, right? Please set the `rule` and `min`.